neqo-transport/src/connection/tests/idle.rs

```diff
@@ -709,21 +705,20 @@ fn keep_alive_with_ack_eliciting_packet_lost() {
     let retransmit = client.process_output(now).dgram();
     assert!(retransmit.is_some());
 
-    // The next callback should be for an idle PING.
-    assert_eq!(
-        client.process_output(now).callback(),
-        IDLE_TIMEOUT / 2 - pto
-    );
+    // The timeout is the twice the PTO, because we've already sent one probe.
+    assert_eq!(client.process_output(now).callback(), pto * 2,);
 
-    // Wait that long and the client should send a PING frame.
+    // Wait for half the idle timeout (less the PTO we've already waited)
+    // so that we get a keep-alive. The actual timeout is the smaller of this
+    // duration or PTO*2, whichever is smaller.
     now += IDLE_TIMEOUT / 2 - pto;
     let pings_before = client.stats().frame_tx.ping;
     let ping = client.process_output(now).dgram();
     assert!(ping.is_some());
     assert_eq!(client.stats().frame_tx.ping, pings_before + 1);
 
     // The next callback is for a PTO, the PTO timer is 2 * pto now.
-    assert_eq!(client.process_output(now).callback(), pto * 2);
+    assert_eq!(client.process_output(now).callback(), pto * 2,);
```

Comment:
```suggestion
    assert_eq!(client.process_output(now).callback(), pto * 2);
```

Reply:
Great catch! Thanks.
I am surprised that compiler doesn't complain about this.

Reply:
Macro formatting is still imperfect.  The compiler knows what sorts of symbols need to appear and can sort of guess, but it can't really know whether the macro cares about a trailing comma or not.